dbt_platform_helper/utils/aws.py

```diff
@@ -48,9 +49,10 @@ def get_aws_session_or_abort(aws_profile: str = None) -> boto3.session.Session:
         botocore.exceptions.UnauthorizedSSOTokenError,
         botocore.exceptions.TokenRetrievalError,
         botocore.exceptions.SSOTokenLoadError,
+        botocore.exceptions.NoCredentialsError,
     ):
         click.secho(
-            "The SSO session associated with this profile has expired or is otherwise invalid."
+            "The SSO session associated with this profile has expired, is not set or is otherwise invalid."
```

Comment:
I think the issue here is that the AWS Profile is either not specified, or doesn't exist, not that the SSO session is expired or invalid.

I wonder if we could catch this separately an output a different error message?

Reply:
Indeed. We trying adding a block towards the beginning of the method, so that it would be a showstopper. I am talking about putting this here, on this [Line](https://github.com/uktrade/platform-tools/blob/449fb0c050b5ec879743f1ad747e8ffc97108171/dbt_platform_helper/utils/aws.py#L28).
The problem is that across the unit tests, we abuse patching, which is a minefield for this sort of change.

The idea was to add a "if not aws_profile" --> showstopper / exit(1).

But the way we build our unit tests actually prevents us from doing this with ease. I spent a lot of time navigating patches and mocks, I tried adding the above functionality in a @mock fixture, to no avail. Connor and I spent ages looking at this. 

I think it's something we need to address ASAP Will. But this looked like the better choice all things considering. Testing it on the console after having uset all the direnv gave us the expected error message, too.